packages/fiber/src/three-types.ts

```diff
@@ -1,399 +1,57 @@
-  tubeGeometry: TubeGeometryProps
-  torusKnotGeometry: TorusKnotGeometryProps
-  torusGeometry: TorusGeometryProps
-  sphereGeometry: SphereGeometryProps
-  ringGeometry: RingGeometryProps
-  planeGeometry: PlaneGeometryProps
-  latheGeometry: LatheGeometryProps
-  shapeGeometry: ShapeGeometryProps
-  extrudeGeometry: ExtrudeGeometryProps
-  edgesGeometry: EdgesGeometryProps
-  coneGeometry: ConeGeometryProps
-  cylinderGeometry: CylinderGeometryProps
-  circleGeometry: CircleGeometryProps
-  boxGeometry: BoxGeometryProps
-  capsuleGeometry: CapsuleGeometryProps
-
-  // materials
-  material: MaterialProps
-  shadowMaterial: ShadowMaterialProps
-  spriteMaterial: SpriteMaterialProps
-  rawShaderMaterial: RawShaderMaterialProps
-  shaderMaterial: ShaderMaterialProps
-  pointsMaterial: PointsMaterialProps
-  meshPhysicalMaterial: MeshPhysicalMaterialProps
-  meshStandardMaterial: MeshStandardMaterialProps
-  meshPhongMaterial: MeshPhongMaterialProps
-  meshToonMaterial: MeshToonMaterialProps
-  meshNormalMaterial: MeshNormalMaterialProps
-  meshLambertMaterial: MeshLambertMaterialProps
-  meshDepthMaterial: MeshDepthMaterialProps
-  meshDistanceMaterial: MeshDistanceMaterialProps
-  meshBasicMaterial: MeshBasicMaterialProps
-  meshMatcapMaterial: MeshMatcapMaterialProps
-  lineDashedMaterial: LineDashedMaterialProps
-  lineBasicMaterial: LineBasicMaterialProps
-
-  // primitive
-  primitive: PrimitiveProps
-
-  // lights and other
-  light: LightProps
-  spotLightShadow: SpotLightShadowProps
-  spotLight: SpotLightProps
-  pointLight: PointLightProps
-  rectAreaLight: RectAreaLightProps
-  hemisphereLight: HemisphereLightProps
-  directionalLightShadow: DirectionalLightShadowProps
-  directionalLight: DirectionalLightProps
-  ambientLight: AmbientLightProps
-  lightShadow: LightShadowProps
-  ambientLightProbe: AmbientLightProbeProps
-  hemisphereLightProbe: HemisphereLightProbeProps
-  lightProbe: LightProbeProps
-
-  // helpers
-  spotLightHelper: SpotLightHelperProps
-  skeletonHelper: SkeletonHelperProps
-  pointLightHelper: PointLightHelperProps
-  hemisphereLightHelper: HemisphereLightHelperProps
-  gridHelper: GridHelperProps
-  polarGridHelper: PolarGridHelperProps
-  directionalLightHelper: DirectionalLightHelperProps
-  cameraHelper: CameraHelperProps
-  boxHelper: BoxHelperProps
-  box3Helper: Box3HelperProps
-  planeHelper: PlaneHelperProps
-  arrowHelper: ArrowHelperProps
-  axesHelper: AxesHelperProps
-
-  // textures
-  texture: TextureProps
-  videoTexture: VideoTextureProps
-  dataTexture: DataTextureProps
-  dataTexture3D: DataTexture3DProps
-  compressedTexture: CompressedTextureProps
-  cubeTexture: CubeTextureProps
-  canvasTexture: CanvasTextureProps
-  depthTexture: DepthTextureProps
+type ThreeExports = typeof THREE
+type ThreeElementsImpl = {
+  [K in keyof ThreeExports as Uncapitalize<K>]: ThreeExports[K] extends ConstructorRepresentation
+    ? Node<ThreeExports[K]>
+    : never
+}
```

Comment:
Thoughts on renaming `Node` to `ThreeElement`? I think we can be more explicit and sidestep clashing with DOM types.

Reply:
I definitely agree, `Node` is too generic.

Reply:
Renamed to `ThreeElement`. Also dropped the `ReactThreeFiber` type namespace since elements' types are no longer hardcoded.